### ISSUE TITLE
KeyGen512 is not defined on browser

### ISSUE THREAD

**Issue by sebmeister2077** [antontutoveanu/crystals-kyber-javascript#11]
I have a problem on the client side and can't seem to use the package. I am using Qwik and whenever i import the library i get this error

![image](https://github.com/antontutoveanu/crystals-kyber-javascript/assets/68665394/3c57e945-ba2f-45c7-b8d4-66e71efa9ae2)

**Comment by ghost** [comment]
I'm not familiar with Qwik, but to get this working in a Vite project I had to pop open node modules, cut out the portion of the code I needed (basically `KeyGen768`, `Encrypt768 `, etc., and the related helper functions). Then I had to add packages that were missing from the client side. Specifically, a package for SHA/SHAKE and a package for buffer. Also, I had to export each of the functions so that I could import them properly. As far as I can tell, this package seems to be designed to run in a server side node environment, so you'll have to adapt it. 

It's hard to say what your specific issue is without any code, but hopefully this gives you some insight. If you can show me some code, I may be able to help more.

**Comment by sebmeister2077** [comment]
Thank you for your reply. Sounds like a lot of trouble, but i found this new package that came out a few days ago and used it on my front end. https://www.npmjs.com/package/crystals-kyber-js

**Comment by ghost** [comment]
Yes, it was a lot of work. Glad you found something that works for you and thanks for the link to the other package. Looks interesting.
